kb/deadline-exceeded-troubleshooting.md

```diff
@@ -0,0 +1,130 @@
+
+```
+
+and
+
+```
+
+./grpc-health-probe -addr=frontendAddress:frontendPort -service=temporal.api.workflowservice.v1.WorkflowService
+
+./grpc-health-probe -addr=matchingAddress:matchingPort -service=temporal.api.workflowservice.v1.MatchingService
+
+./grpc-health-probe -addr=historyAddress:historyPort -service=temporal.api.workflowservice.v1.HistoryService
+
+```
+
+- [History Service](/clusters#history-service)
+- [Persistence](/clusters#persistence) database
+- [Advanced Visibility](/visibility#advanced-visibility) database
+
+Logs can also be used to find Client [Query](/workflows#queries) requests that failed with this error.
+
+#### Check your Cluster metrics
+
+Cluster metrics can be used to detect issues (such as 'resource exhausted') that implact cluster health.
+`Resource exhausted` errors can cause your client to fail and send a `Deadline exceeded` error.
+
+Use the command below to check for errors in `RpsLimit`, `ConcurrentLimit` and `SystemOverloaded`.
+
+```
+
+sum(rate(service_errors_resource_exhausted{}[1m])) by (resource_exhausted_cause)
+
+```
+
+Look for high latencies, short timeouts, and other abnormal [Cluster metrics](/references/cluster-metrics).
+If the metrics come from a specific service (such as History Service), check that the service is connected.
+
+#### Check Workflow logic
+
+Check your [Client and Worker configuration](/references/configuration) files for missing or invalid target values, such as:
+
+- Server names
+- Network or host addresses
+- Certificates
+
+Invalid targets also cause `connection refused` errors alongside `deadline exceeded`.
+Check that the Client connects after updating your files.
+
+## Advanced troubleshooting
+
+If you're still getting this error, here are further things to check.
+
+#### After enabling mTLS
+
+Check the health of the Cluster with `tctl cluster health`.
+
+```
+tctl --address [SERVER_ADDRESS] cluster health
+```
+
+Add any missing [environment variables](/references/web-ui-environment-variables) to the configuration files, and correct any incorrect values.
+Server names and certificates must match between Frontend and internode.
+
+#### After restarting the Temporal Cluster
+
+You might not be giving the Cluster enough time to respond and reconnect.
+Restart the Server, wait, and then check all services for connectivity and further errors.
+
+If the error persists, review your Workflow Execution History and server logs for more specific causes before continuing to troubleshoot.
+
+#### When executing or scheduling Workflows
+
+One or more services may be unable to connect to the [Frontend Service](/clusters#frontend-service).
+The Workflow might be unable to complete requests within the given connection time.
+
+Increase the value of `frontend.keepAliveMaxConnectionAge` so that requests can be finished before the connection terminates.
+
+::: note
+
+If you increase `frontend.keepAliveMaxConnectionAge` values, consider checking for server overload.
```

Comment:
Think users will ask "how to check for server overload" here. Wdyt?

Reply:
Just changed to "monitor server performance."